In [30]:
import pandas as pd
import numpy as np

## 1.The experiment shows that the average monthly electricity consumption for a house is 458 units. The study records a total of 3206 units of electricity consumption. Calculate the duration of the experiment. Also, identify the criteria to calculate the median consumption. 


In [2]:
# set the average electricity consumption in 'x_bar'
x_bar = 458

# set the total electricity consumption in 'sum_x'
sum_x = 3206

# the average/mean is calculated as the ratio of total consumption to the number of recorded months (n)
# thus, 'n' can be calculated as the ration of 'sum_x' to 'x_bar'
n = sum_x / x_bar

# print the number of months
print('Duration of the experiment:', n, 'months')

Duration of the experiment: 7.0 months


As n = 7 (odd number), the median consumption is the middlemost observation when the data is arranged in the ascending or descending order.

## 2. Find the wage above which 65% of wages would lie.
     

            Consider the wages (in dollars) of 15 workers as given below:

            wage = [247, 228, 352, 298, 345, 312, 300, 285, 246, 352, 297, 304, 265, 325, 276]

In [3]:
# consider the series of wages
wage = pd.Series([247, 228, 352, 298, 345, 312, 300, 285, 246, 352, 297, 304, 265, 325, 276])

# compute 35th percentile using 'quantile'
# As 35th percentile divides the data in 35-65%
wage.quantile(0.35)

284.1

65% of wages are above 284.1 dollars.

## 3. Jimmy requires 38 minutes on average, with a standard deviation of 5 minutes to distribute the newspapers. On the other hand, Richard requires 27 minutes on average, with a standard deviation of 6 minutes to distribute the newspapers. Identify most consistent distributor.


In [4]:
# average time for Jimmy 
jimmy_avg = 38

# standard deviation for Jimmy
jimmy_std = 5

# average time for Richard 
richard_avg = 27

# standard deviation for Richard
richard_std = 6

# calculate coefficient of variation for both the distributors
jimmy_cv = (jimmy_std/jimmy_avg)*100
richard_cv = (richard_std/richard_avg)*100

# print the cv 
print('C.V. for Jimmy:', jimmy_cv)
print('C.V. for Richard:', richard_cv)

C.V. for Jimmy: 13.157894736842104
C.V. for Richard: 22.22222222222222


## 4. Identify the extreme weights from the given list of weights using the interquartile range.
Consider the weights (in kg) of 10
students as given below:
weight = [14.7, 15.2, 39.8, 44.5, 21.2,
28.5, 35.2, 19.7, 40.4, 16.5]


In [5]:
# consider a series of given weights
weight = pd.Series([44.7, 55.2, 49.8, 44.5, 15.2, 78.5, 75.2, 19.7, 50.4, 46.5])

# calculate the first quartile (Q1)
Q1 = weight.quantile(0.25)

# calculate the third quartile (Q3)
Q3 = weight.quantile(0.75)

# calculate the interquartile range (IQR)
IQR = Q3 - Q1

# retrieve the outliers
# pass the condition to detect the outliers
# '|' is used as 'OR' operator on multiple conditions   
outliers = weight[(weight < (Q1 - 1.5 * IQR)) | (weight > (Q3 + 1.5 * IQR))]

# print the weights
outliers

4    15.2
5    78.5
6    75.2
7    19.7
dtype: float64

## 5. A car showroom in Ohio exhibits a total of 12 cars of different brands. Identify the type of skewness in the showroom prices of cars. 


#### Consider the prices (in dollars) for 12 cars as given below:
 price = [14751, 16422, 15398, 9445, 12589, 11687, 10692, 8475, 11184, 9961, 12898, 11905]

In [29]:
# consider the series of car prices
price = pd.Series([14751, 16420, 15398, 9445, 12589, 11687, 10692, 8775, 11184, 9961, 12898, 11905])
# calculate mean
mean = np.mean(price)

# calculate median
median = np.median(price)

# calculate standard deviation
std = np.std(price)

# calculate the coefficient of skewness
skewness = (3.0*(mean - median))/std

# print the value of skewness
print('Coefficient of Skewness:', skewness)

Coefficient of Skewness: 0.45265633664561267


## 6. A company offers its customers to take a vehicle and an eduacational loan. A study shows that the probability that a customer takes a vehicle loan is 0.35, the probability that a customer takes an educational loan is 0.46 and the probability that a customer takes at least one of the loan is 0.59. Find the probability that a customer takes both vehicle and educational loans.


In [3]:
prob_vehicle = 0.35

# consider an event B: Customer takes an educational loan
# probability that a customer takes an educational loan
prob_edu = 0.46

# probability that a customer takes one of the loans (P(A ⋃ B))
prob_vehicle_or_edu = 0.59
prob_vehicle_and_edu = prob_vehicle + prob_edu - prob_vehicle_or_edu

print('The probability that a customer takes both the loans:', prob_vehicle_and_edu)

The probability that a customer takes both the loans: 0.22000000000000008


## 7. Consider two independent events X and Y. If the probability of occurrence of event X is 0.4 and the probability of occurrence of at least one event is 0.6. Find the probability of occurrence of event Y. 


In [8]:
# probability of occurrence of event X
prob_X = 0.4

# probability of occurrence of at least one event
prob_X_or_Y = 0.6
prob_Y = (prob_X_or_Y - prob_X) / (1 - prob_X)

print('Thr probability of occurrence of event Y:', prob_Y)

Thr probability of occurrence of event Y: 0.33333333333333326


## 8. Consider a vending machine containing 6 boxes each of four different juice flavors (Mango, Apple, Guava and Kiwi). Find the probability that the juice box picked is either of Mango or Guava flavor.                 

In [4]:
boxes = 6
total_boxes = 6*4
prob_mango = boxes / total_boxes
prob_guava = boxes / total_boxes

req_prob = prob_mango + prob_guava

print('The probability that the juice box picked is either of Mango or Guava flavor:', req_prob)

The probability that the juice box picked is either of Mango or Guava flavor: 0.5


## 9. Consider a student appearing for 3 different exams either pass or fail in each exam. Let a marginal probability of a student passing the Economics, Mathematics and German exam be 0.67. The probability that the student fails in the Economics exam is 0.10 and the probability that the student fails in the German exam is 0.09. Find the probability that a student fails in the Mathematics exam.


Consider a discrete variable `X` denoting the passing or failing of an exam and a discrete variable `Y` denoting the three subjects of the exam. The sum of marginal probabilities of X (or Y) should be always equal to 1.

In [5]:
prob_X_pass = 0.67
prob_X_fail = 1 - prob_X_pass
prob_eco_fail = 0.10
prob_german_fail = 0.09
prob_maths_fail = prob_X_fail - prob_eco_fail - prob_german_fail 

print('The probability that a student fails in Mathematics exam', prob_maths_fail)

The probability that a student fails in Mathematics exam 0.13999999999999996


## 10. The probabilities that a machine will produce 2, 5, 7, 9 defective batteries in a day are 0.3, 0.22, 0.15, 0.1 respectively. Find the expected number of defective batteries produced in a day.


In [19]:
X = np.array([2, 5, 7, 9])
prob_X = np.array([0.3, 0.22, 0.15, 0.1])

# caclulate the expected number of defective batteries
expected_value = np.sum(X*prob_X)

print('Expected number of defective batteries produced in a day:', expected_value)

NameError: name 'np' is not defined

## 11. In a pack of 52 cards, what is the probability of selecting a red card?


In [14]:
probability_of_selecting_card = 1/52
required_probability = 26 * probability_of_selecting_card
print('The probability of selecting a red card', required_probability)

The probability of selecting a red card 0.5


## 12. From the experience, it is seen that 4% of the tyres produced by the machine are defective. Out of the 10 tyres produced, find the probability that at most 3 are defective.


In [11]:
from scipy.stats import binom

# X: the number of defective tyres produced 
# n = 10
# p = 0.04
# required probability is P(X ≤ 3) = P(X = 0) + P(X = 1) + P(X = 2) + P(X = 3)
req_prob = binom.pmf(0, 10, 0.04) + binom.pmf(1, 10, 0.04) + binom.pmf(2, 10, 0.04) + binom.pmf(3, 10, 0.04)

print('The probability that at most 3 tyres out of 10 are defective:', req_prob)

The probability that at most 3 tyres out of 10 are defective: 0.9995574006399994


In [12]:
# or one may use the cdf() and pass the k to get P(X ≤ k)
binom.cdf(3, 10, 0.04) 

0.9995574006399999

## 13. The number of trucks crossing a bridge during the day follow a Poisson distribution with mean 18. What is the probability that on a randomly selected day 12 trucks would have crossed the bridge?


In [13]:
# import the required library
from scipy.stats import poisson

# X: number of trucks crossing a bridge during the day
# x = 12
# m = 18
# The required probability is P(X = 12)
req_prob = poisson.pmf(12, 18)

print('The probability that on a randomly selected day 12 trucks would have crossed the bridge:', req_prob)

The probability that on a randomly selected day 12 trucks would have crossed the bridge: 0.03678175293604137


## 14. A random variable follows uniform distribution in the interval 5 to 10. Obtain the mean and variance. 

In [15]:
# lower limit of the uniform distribution 
lower_limit = 5

# upper limit of the uniform distribution 
upper_limit = 10

In [16]:
# compute the mean of the uniform distribution
uniform_mean = (upper_limit - lower_limit) / 2

# print the mean
print('Mean:', uniform_mean)

Mean: 2.5


In [17]:
# compute the variance of the uniform distribution
uniform_variance = (upper_limit - lower_limit)**2 / 12

# print the variance
print('Variance', uniform_variance)

Variance 2.0833333333333335


### 15. Consider an random variable that follows normal distribution with mean 5 and variance 10 obtain
1. P(X < 1)<br>
2. P(X > 5)<br>
3. P(3 < X < 7)


In [25]:
from scipy.stats import norm
mu = 5 
var =10

# standard deviation of the distribution 
sd = np.sqrt(var)

In [26]:
# 1. P(X < 1)
# use cdf() to calculate P(X < k)
norm.cdf(1, mu, sd)

0.1029516053660342

In [27]:
# 2. P(X > 5)
# use cdf() to calculate P(X < k)
1 - norm.cdf(5, mu, sd)

0.5

In [28]:
# 3. P(3 < X < 7)
# use cdf() to calculate P(X < k)
norm.cdf(6, 3, 2) - norm.cdf(4, 3, 2)

0.2417303374571288